src/scripts/developer.ts

```diff
@@ -68,3 +68,104 @@ const resetForm = (developer: Developer) => {
   developer.experience = 0;
   developer.email = "";
 };
+
+//Add Property of developer object
+export const addProperty = <T>(developer: Developer, additionalProps?: T) => {
+  if (additionalProps) {
+    Object.assign(developer, { additionalProperties: additionalProps });
+  }
+  return {
+    ...developer,
+    additionalProperties: additionalProps,
+  };
+};
+
+// Update Developer's Skill
+export const updateSkill = (
+  developers: Developers,
+  developer: Developer,
+  oldSkill: string,
+  newSkill: string
+) => {
+  const currentDeveloper = developers?.find(
+    ({ email }) => email === developer.email
+  );
+
+  if (currentDeveloper) {
+    const index = currentDeveloper?.skills?.indexOf(oldSkill);
+
+    if (index > -1) {
+      currentDeveloper.skills.splice(index, 1, newSkill);
+      developers.splice(
+        developers.findIndex((item) => item === developer),
+        1,
+        currentDeveloper
+      );
+    }
+  }
+};
+
+// Condition for removing a Developer
+export const conditionForRemoving = (developer: Developer) =>
+  developer.age < 18;
+
+// Remove Developer By Condition
+export const removeDeveloperByCondition = (
+  developers: Developers,
+  conditionForRemoving: (developer: Developer) => boolean
+) => {
+  const devToBeRemoved = developers.filter(conditionForRemoving);
+  devToBeRemoved.forEach((item) =>
+    developers.splice(developers.indexOf(item), 1)
+  );
+};
+
+// Sort Developer by Employment and Age
+export const sortDeveloperByEmploymentAndAge = (developers: Developers) => {
+  console.log(
+    developers.sort((a, b) => {
```

Comment:
do not use console log inside this function, instead, just return the sorted devs from this function

function should not be concerned with printing, its job is to sort the devs